src/fontra/views/fontinfo/panel-development-status-definitions.js

```diff
@@ -0,0 +1,362 @@
+      this.setupUI();
+    }
+  }
+}
+
+function getStatusFieldDefinitions(fontController) {
+  const statusFieldDefinitions =
+    fontController.customData["fontra.sourceStatusFieldDefinitions"];
+  if (statusFieldDefinitions) {
+    return statusFieldDefinitions;
+  }
+  return [];
+}
+
+addStyleSheet(`
+.fontra-ui-font-info-status-definitions-panel-status-def-box {
+  background-color: var(--ui-element-background-color);
+  border-radius: 0.5em;
+  padding: 1em;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: max-content max-content max-content max-content auto;
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-value {
+  width: 2em;
+  text-align: center;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-delete {
+  justify-self: end;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-color-input {
+  width: 8em;
+  /*height: 1.6em; without height: best compromise for all browsers*/
+  margin: 0;
+  padding: 0;
+  outline: none;
+  border: none;
+  border-color: transparent;
+  border-radius: 0.25em;
+}
+`);
+
+class StatusDefinitionBox extends HTMLElement {
+  constructor(fontController, statusIndex, postChange, setupUI) {
+    super();
+    this.classList.add("fontra-ui-font-info-status-definitions-panel-status-def-box");
+    this.fontController = fontController;
+    this.statusIndex = statusIndex;
+    this.postChange = postChange;
+    this.setupUI = setupUI;
+    this._updateContents();
+  }
+
+  get statusDef() {
+    return this.fontController.customData["fontra.sourceStatusFieldDefinitions"][
+      this.statusIndex
+    ];
+  }
+
+  doChecks(newStatusDef) {
+    const statusDefinitions =
+      this.fontController.customData["fontra.sourceStatusFieldDefinitions"];
+    if (statusDefinitions.some((statusDef) => statusDef.value == newStatusDef.value)) {
+      message(
+        `Can’t edit status definition value`,
+        `“${newStatusDef.value}” exists already, please use a different value.`
+      );
+      return false;
+    }
+    return true;
+  }
+
+  replaceStatusDef(newStatusDef, undoLabel, statusIndex = this.statusIndex) {
+    if (!this.doChecks(newStatusDef)) {
+      this.setupUI();
+      return;
+    }
+
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"][statusIndex] =
+        newStatusDef;
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+
+  deleteStatusDef(statusIndex) {
+    const undoLabel = `delete status def '${this.statusDef.name}'`;
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"].splice(statusIndex, 1);
```

Comment:
If the list gets empty as the result of this, the whole customData entry should be deleted.

Reply:
Not sure if this is related to that, but I am wondering if we want to update/modify the status definitions for each glyph source as well? For example, when editing the status definition value, do we want to update the value in all glyph sources? And if we delete one status definition, do we want to delete this value in all glyph sources?

Reply:
> but I am wondering if we want to update/modify the status definitions for each glyph source as well? 

No, that is completely unfeasible. Imagine a font with 40000 glyphs which is stored remotely in a database.

Reply:
OK, in that case I am closing this.